lib/cog/commands/filter.ex

```diff
@@ -57,6 +92,29 @@ defmodule Cog.Commands.Filter do
   defp maybe_pluck(item, _),
     do: item
 
+  defp build_path([], [], path), do: path
+  defp build_path([key|remaining], acc, path) do
```

Comment:
Any way to split the path using a regular expression instead? This code is a bit opaque.